### PR TITLE
Metrics to Main

### PR DESCRIPTION
**Pull request by cijothomas**
We are prepping for 1.2.0-alpha1 release. 1.2.0 is the next planned release, and now we can work in main branch for metrics code as well.

### CODE REVIEW

File: build/Common.prod.props [hunk 1]
```diff
@@ -6,9 +6,6 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="$(MicrosoftCodeAnalysisAnalyzersPkgVer)" Condition=" $(OS) == 'Windows_NT'">
-      <PrivateAssets>All</PrivateAssets>
-    </PackageReference>
```
Comment (alanwest) [inline build/Common.prod.props:11]: We will continue to leave this disabled on main for now?

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,9 @@ branch](https://github.com/open-telemetry/opentelemetry-dotnet/tree/metrics),
 please check the latest changes
 [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/metrics/src/OpenTelemetry.Api/CHANGELOG.md#experimental---metrics).
 
+* Removed existing Metrics code as the spec is completely being re-written.
+  ([#2030](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2030))
+
```
Comment (alanwest) [inline src/OpenTelemetry.Api/CHANGELOG.md:12]: Might make sense to update this changelog to reflect the new state of things.

File: src/OpenTelemetry/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,9 @@ branch](https://github.com/open-telemetry/opentelemetry-dotnet/tree/metrics),
 please check the latest changes
 [here](https://github.com/open-telemetry/opentelemetry-dotnet/blob/metrics/src/OpenTelemetry/CHANGELOG.md#experimental---metrics).
 
+* Removed existing Metrics code as the spec is completely being re-written.
+  ([#2030](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2030))
```
Comment (alanwest) [inline src/OpenTelemetry/CHANGELOG.md:11]: Same thing for the SDK changelog

File: test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs [hunk 4]
```diff
@@ -34,7 +34,7 @@ public TracerProviderSdkTest()
             Activity.DefaultIdFormat = ActivityIdFormat.W3C;
         }
 
-        [Fact]
+        [Fact(Skip = "Get around GitHub failure")]
```
Comment (alanwest) [inline test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs:37]: Is it still necessary to skip this check on the main branch?

**Review by alanwest — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Api/CHANGELOG.md [hunk 5]
```diff
@@ -81,9 +84,6 @@ Released 2021-Jan-29
 * `Status.WithDescription` will now ignore the provided description if the
   `Status.StatusCode` is anything other than `ERROR`.
   ([#1655](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1655))
-* Metrics removed as it is not part 1.0.0 release. See issue
```
Comment (reyang) [inline src/OpenTelemetry.Api/CHANGELOG.md:84]: I guess we need to keep this.

**Review by reyang — APPROVED** [review]
LGTM. After this PR got merged, might need to update the CI definition by removing metrics branch.

**Pull request merged**